### PR TITLE
prepare header propagation configuration for #1284

### PR DESCRIPTION
**Pull request by bnjjj**
Adds `request` subsection into header propagation configuration to prepare for #1284 coming post 1.0

```patch
all:
+     request:
        - remove:
            named: "test"
```

### CODE REVIEW

File: apollo-router/src/plugins/headers.rs [hunk 1]
```diff
@@ -37,6 +37,15 @@ use crate::SubgraphRequest;
 
 register_plugin!("apollo", "headers", Headers);
 
+#[derive(Clone, JsonSchema, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+struct HeadersLocation {
```
Comment (garypen) [inline apollo-router/src/plugins/headers.rs:42]: I think we need to mark this as being non-exhaustive:

```
#[non_exhaustive]
```

**Review by garypen — CHANGES_REQUESTED** [review]
Looks good. Needs a breaking changelog and I added a comment about marking the new struct as non-exhaustive.

File: apollo-router/src/plugins/headers.rs (hunk 1 above)
Comment (bnjjj) [reply apollo-router/src/plugins/headers.rs:42]: Is it important to mark it as `non_exhaustive` even if the struct is private ?

Comment (garypen) [reply apollo-router/src/plugins/headers.rs:42]: No it isn't. Ignore me.

File: NEXT_CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,22 @@ By [@USERNAME](https://github.com/USERNAME) in https://github.com/apollographql/
 # [x.x.x] (unreleased) - 2022-mm-dd
 
 ## ❗ BREAKING ❗
+
+### Respect supergraph path for kubernetes deployment probes [PR #1795](https://github.com/apollographql/router/pull/1795)
```
Comment (garypen) [inline NEXT_CHANGELOG.md:30]: mis-merge or ?

**Review by garypen — CHANGES_REQUESTED** [review]

File: NEXT_CHANGELOG.md (hunk 2 above)
Comment (bnjjj) [reply NEXT_CHANGELOG.md:30]: Oh yes, I tried to use the "update branch" from github but probably a bad idea :p

Comment (bnjjj) [reply NEXT_CHANGELOG.md:30]: Oh no, it's a bad copy/paste, my bad

Comment (bnjjj) [reply NEXT_CHANGELOG.md]: fixed

**Review by garypen — APPROVED** [review]
Nice work. Thanks!

**Review by o0Ignition0o — APPROVED** [review]

**Review by BrynCooke — APPROVED** [review]

**Pull request merged**